docs/chunk-deduplication.md

```diff
@@ -0,0 +1,116 @@
+﻿# Probntroduction
+In container images, there are often a large number of duplicate files or content, and these duplicate parts occupy a large amount of storage space, especially in high-density deployment scenarios. As the number of Nydus images grows, it will bring many problems such as low storage space utilization and excessive consumption of bandwidth resources. To do this, an effective deduplication mechanism (deduplication) needs to be designed to solve this problem.
+
+Unlike traditional OCI, which distributes images at a layer-granular level, the smallest unit of a Nydus image is a chunk, so the deduplication algorithm needs to be deduplicated in chunk units. At the same time, we want to deduplicate multiple aspects of the Nydus image, including between Nydus images and between different versions of the same Nydus image. No matter which deduplication method is essentially to deduplicate the repeated chunks in the image, only one duplicate chunk is retained, and the reference to the chunk is used instead of other duplicate chunks to reduce the storage space occupation, so as to maximize the data transmission and storage capabilities of Nydus and improve the access speed and efficiency of the image.
+# General idea
+The deduplication algorithm first needs to select the duplicate chunk in the image according to the image information such as the number of occurrences of chunk, chunk size, chunk image to which the chunk belongs and the corresponding version, and generate chunkdict, chunkdict records the unique identifier or fingerprint of chunk, only need to store chunkdict, other images can refer to chunk in chunkdict by reference.
+
+The deduplication algorithm is divided into two parts, the first part is the DBSCAN clustering algorithm, which deduplicates different images; The second part is the exponential smoothing algorithm, which deduplicates different versions within the image.
+
+**The general process is as follows:**
+1. Store the image information to the local database,
+2. Extract the image information and call the DBSCAN clustering algorithm to deduplicate different images.
+3. Deduplicate the dictionary content in 2, and call the exponential smoothing algorithm for each image separately for image version deduplication.
+4. Get the deduplication dictionary generated by running the two algorithms and drop the disk.
+# Algorithm detailed process
+## Overall  Input
+
+```shell
+nydusify chunkdict generate --sources \
+	localhost:5000:redis:nydus_7.0.1, \
+	localhost:5000:redis:nydus_7.0.2，\
+	localhost:5000:redis:nydus_7.0.3 \ 
+```
+*** 
+`nydusify chunkdict generate` calls two commands `nydus-image  chunkdict save` and `nydus-image  chunkdict generate`  to store image information into the database and generate a list of chunks to be deduplicated
```

Comment:
The doc need to be updated.

Reply:
The nydusify command involved in the doc is being further refined and will be updated in the future